### PR TITLE
 Add syncLdapUser param to security_config

### LINKED ISSUE #510: Missing rundeck.security.syncLdapUser config setting
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 7
- Ruby: 2.7
- Distribution: 
- Module version: 8.0.0

## How to reproduce (e.g Puppet code you use)

Trying to set rundeck security setting syncLdapUser the setting is not setted

```
class { 'rundeck':
 security_config = {
    'syncLdapUser' => true,
  }
}
```
## What are you seeing

The setting is missing on rundeck-config.groovy file

## What behaviour did you expect instead

The setting being setted on rundeck-config.groovy file

### PR DESCRIPTION
**Pull request by nosrio**
#### Pull Request (PR) description

Add `rundeck.security.syncLdapUser` to  rundeck_config.groovy file. This flag enable using LDAP user attributes to update the email, first name, and last name properties of your Rundeck users. 
Add ldap user attributes used to source the email, and name properties.
Remove un used templates.

ref: https://docs.rundeck.com/docs/administration/security/authentication.html#ldap

#### This Pull Request (PR) fixes the following issues

Fixes #510 
Fixes #416

### CODE REVIEW

**Comment by nosrio** [comment]
Adding some tests.

**Comment by kenyon** [comment]
> `Failure/Error: apply_manifest(pp, catch_changes: true)`

This is the second run, to check for idempotency. Something is changing during that run.

**Comment by nosrio** [comment]
I believe that these checks started failling when #502 was merged. In fact, that PR do not have all checks passed.

I am triying to debug why the second run is not idempotent but it is not related to the changes on thisI think.

**Comment by nosrio** [comment]
@bastelfreak can you help me to solve this errors?

**Comment by nosrio** [comment]
@bastelfreak please review

File: templates/_auth_ldap.epp [hunk 1]
```diff
@@ -21,6 +21,11 @@ com.dtolabs.rundeck.jetty.jaas.<%= $rundeck::config::ldap_login_module %> <%= $r
   userRdnAttribute="<%= $rundeck::config::auth_config['ldap']['user_rdn_attribute'] %>"
   userIdAttribute="<%= $rundeck::config::auth_config['ldap']['user_id_attribute'] %>"
   userPasswordAttribute="<%= $rundeck::config::auth_config['ldap']['user_password_attribute'] %>"
+<%- if 'syncLdapUser' in $rundeck::config::security_config.keys and $rundeck::config::security_config['syncLdapUser'] == true {-%>
```
Comment (bastelfreak) [inline templates/_auth_ldap.epp:24]: same here

**Review by bastelfreak — COMMENTED** [review]
please take a look at the inline comments i did and squash your commits together.

**Comment by nosrio** [comment]
Done @bastelfreak :)

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**